BugHunter101

medium

# withdraw() using unsafe function "call()"

## Summary

In FootiumEscrow.sol, the withdraw() using unsafe function "call()"

## Vulnerability Detail

In FootiumEscrow.sol, the withdraw() using unsafe function "call()" , and it does not use modify nonReentrant

## Impact

This is an unsafe way of coding and may cause potential security issues

## Code Snippet

https://github.com/sherlock-audit/2023-04-footium/blob/main/footium-eth-shareable/contracts/FootiumEscrow.sol#L132

function withdraw() external onlyClubOwner {
        uint256 balance = address(this).balance;
        if (balance > 0) {
            (bool sent, ) = payable(msg.sender).call{value: balance}("");
            if (!sent) {
                revert FailedToSendETH(balance);
            }
        }
    }

## Tool used

Manual Review

## Recommendation

using transfer() or use modify nonReentrant.
